lib/packwerk/const_node_inspector.rb

```diff
@@ -10,13 +10,11 @@ class ConstNodeInspector
 
     def constant_name_from_node(node, ancestors:)
       return nil unless Node.constant?(node)
-
       # Only process the root `const` node for namespaced constant references. For example, in the
       # reference `Spam::Eggs::Thing`, we only process the const node associated with `Spam`.
-      parent = ancestors.first
-      return nil if parent && Node.constant?(parent)
+      return nil if root_constant?(ancestors)
 
-      if constant_in_module_or_class_definition?(node, parent: parent)
+      if constant_in_module_or_class_definition?(node, ancestors: ancestors)
```

Comment:
I don’t think this part of the change is worth it, because we’re now intentionally passing more information into `constant_in_module_or_class_definition?` than it needs to do its job. For the caller it’s useful to be able to see at a glance that this decision depends only upon the parent node, not the whole ancestor chain, and for the implementer it’s helpful to be prevented from accidentally poking around in the rest of the ancestor chain unless you really mean it (and deliberately change the arguments to reflect that). WDYT?

Reply:
Agreed, I'm providing too much in the private method and making the intention behind the code harder to understand 👍